### PR TITLE
[OLD, check other PR] out_oms_blob.rb: Fix both http 400 and event duplication

### PR DESCRIPTION
**Pull request by abenbachir**
This is a really nasty bug.
This PR fixes 2 bugs:
Bug 1: Getting http 400 erros due thread concurrency
Bug 2: Duplicate event were uploaded when the 2nd tag upload had failed and scheduled fore retry. During retry fluentd will call write() again with the whole chunk including both tags.

### CODE REVIEW

**Review by nidhanda — APPROVED** [review]

File: source/code/plugins/out_oms_blob.rb [hunk 1]
```diff
@@ -348,32 +351,31 @@ def handle_record(tag, records)
         raise "The tag does not have at least 4 parts #{tag}"
       end
 
-      start = Time.now
-      blob_uri, blocks_committed, blob_size = get_blob_uri_and_committed_blocks(container_type, data_type, custom_data_type, suffix)
-      time = Time.now - start
-      @log.debug "Success getting the BLOB information in #{time.round(3)}s"
+	  # get a lock before getting the blob information and blob append to avoid 400 http storage errors when parallel threads are writing
+	  @log.trace "[#{Thread.current.object_id}] Trying to enter critical section before append_blob()"
+	  BlockLock.lock
```
[the remark this answers is not in the document]
Comment (abenbachir) [reply source/code/plugins/out_oms_blob.rb:356]: yes, the lock was already but we have also moved "Getting the BLOB information" inside the block so each thread will get latest committed block list before going to upload the new block.

**Review by canfikret — COMMENTED** [review]
@abenbachir , let's drive this to conclusion. Thanks.

**Review by canfikret — APPROVED** [review]

**Comment by abenbachir** [comment]
I found a better way to solve this issue in this PR #1241 
I think I will cancel this PR and focus on that one.

**Comment by hestolz** [comment]
Thank you for your submission. As [previously announced](https://azure.microsoft.com/en-us/updates/were-retiring-the-log-analytics-agent-in-azure-monitor-on-31-august-2024/), the Log Analytics agent has been **deprecatated** and has no support as of **August 31, 2024.** If you use the Log Analytics agent to ingest data to Azure Monitor, [migrate now to the new Azure Monitor agent](https://docs.microsoft.com/en-us/azure/azure-monitor/agents/azure-monitor-agent-migration). As part of repo archival, open issues and pull requests will be closed.

**Pull request closed without merging**